### ISSUE TITLE
Store should allow writing version-less streams

### ISSUE THREAD

**Issue by TomKaltz** [Rotorsoft/eventually-monorepo#84]
For streams that do not require strict versioning and to provide more write reliability (namely less concurrency conflict for higher throughput generic sub-streams) the event store should have the ability to write to streams without versioning of events.  The current PG event store implementation does not enforce contiguous event versioning for non-reducible streams but there is a database constraint that currently blocks this usage [here](https://github.com/Rotorsoft/eventually-monorepo/blob/432f498c12fd0c11724ec673c65fc8fd989ba5b5/libs/eventually-pg/src/seed.ts#L45).

**Comment by Rotorsoft** [comment]
Yes, these are part of the “technical” optimizations I mentioned in the other issue, and I think we'll need a separate store for external streams for two reasons:
- To achieve higher throughput with simpler logic, fewer indexes, no strict ordering guarantees, etc.
- These streams shouldn’t be part of the main “all” stream representing actual business events. 
 
Once external events are ingested via integration policies, they become part of the “ordered” ALL stream (a slower but more robust process)

The question is how we should structure these external stream stores: should we have a single store supporting multiple streams, or separate stores similar to how projection stores are handled? This is the direction I’m exploring with integration utilities and optimizations, but it might take some time to implement given my current commitments.

**Comment by TomKaltz** [comment]
Since we are trying to remove external systems as much as possible from the domain, a separate store/table for these sub-streams is probably best.

**Comment by Rotorsoft** [comment]
See the latest commits and the introduction of a message queue abstraction to resolve these types of integration problems:
https://github.com/Rotorsoft/eventually-monorepo/blob/master/libs/eventually-pg/src/PostgresMessageQueue.ts
